### ISSUE TITLE
Update column names to be more informative

### ISSUE THREAD

**Issue by jchodera** [MobleyLab/FreeSolv#33]
Now that we're distributing FreeSolv in multiple forms (SDF files, Orion datasets), it would be useful if the column names (that become SD tags) were more informative and self-documenting about things like units. For example, the "expt" label is problematic since (1) it doesn't describe *which* of the many properties this is the experimental value for, and (2) it doesn't give the units. More informative names would very much help!

**Comment by davidlmobley** [comment]
Agreed! What's your favorite way to attach units now? `simtk.unit`, or have you migrated towards something else? 

Though, there's also the issue of how things will pickle or go to JSON which I've not thought through yet. Perhaps we should just embed units in the name of the key or tag.

**Comment by jchodera** [comment]
We're still using `simtk.unit` for Python, which can also render units in a human-readable form.

@bas-rustenburg has used [pint](https://pint.readthedocs.io), which is very similar and independent of OpenMM.
